test/can_obd2/can_mapping_test.cpp

```diff
@@ -91,8 +92,22 @@ void CANMappingTest::extract_test_bit_mode(void)
                                         test_value = 0x5555555555555555 & (((uint64_t)1 << length) - 1);
                                 }
 
+                                uint64_t encoded_value = test_value;
+                                if (endian == 0) {
```

Comment:
Does this even matter if the bit-pattern since the bit-pattern is the same all the way through (little endian vs big endian doen't matter when the pattern is the same for each byte).

Reply:
yeah, we have to craft the bit pattern so it looks like this for little endian.  (xxx means ignore bits)

LLLLLLLL HHHHxxxx

Reply:
I am aware of that.  What i mean is that the pattern 0x5555 looks the same in both little and big endian.  Look at it on a hex editor.  Point is that you shouldn't use a pattern like that because it can mask bugs.  Instead you should use a pattern like 0xAA55.  That pattern flips the bits per byte (meaning each byte is the negative of the other).  This will more likely expose problems.

Reply:
Got it.  Ok, I changed the bit pattern to 0xAA55AA55AA55...